pkg/volume/csi/csi_mounter.go

```diff
@@ -90,9 +109,29 @@ func (c *csiMountMgr) SetUpAt(dir string, fsGroup *int64) error {
 	nodeName := string(c.plugin.host.GetNodeName())
 	attachID := getAttachmentName(csiSource.VolumeHandle, csiSource.Driver, nodeName)
 
+	// persist volume info data for teardown
+	volData := map[string]string{
+		volDataKey.specVolID:    c.spec.Name(),
+		volDataKey.volHandle:    csiSource.VolumeHandle,
+		volDataKey.driverName:   csiSource.Driver,
+		volDataKey.nodeName:     nodeName,
+		volDataKey.attachmentID: attachID,
+	}
+
+	if err := saveVolumeData(c.plugin, c.podUID, c.spec.Name(), volData); err != nil {
```

Comment:
We should do this immediately after `os.MkdirAll( ...)` below so that the `saveVolumeData` doesn't have to create a dir.

Also if the mount operation fails after this, we should cleanup the saved data. But we have to be careful, if the mount exists (maybe from a previous iteration), we need to make sure that we don't accidently delete data before it is unmounted.

Reply:
We should do `saveVolumeData(...)` after `os.MkdirAll` but before `csi.NodePublishVolume` -- so that if a mount succeeds and then the operation fails immediately afterwards, we have enough to cleanup.

Can either move it now or create an issue to clean it up later.